### PR TITLE
done

### PR DESCRIPTION
**Pull request by mikkajimenez**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -10,52 +12,117 @@
 // Reminder: The test will call your function
 // Run the file with the following command: $ yarn jest
 
-
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
+const secretCodeWord1 = "Lackadaisical";
```
Comment (sjproctor) [inline code-challenges.test.js:19]: Remember that your test variables should be inside the scope of your test.

File: code-challenges.test.js [hunk 2]
```diff
@@ -10,52 +12,117 @@
...
+      // replaces Aa, Ee and Ii to 4, 3 and 1
+      return "4";
+    } else if (value === "e" || value === "E") {
+      return "3";
+    } else if (value === "i" || value == "I") {
+      return "1";
+    } else return value;
+  });
+  return strArray.join(""); //joins the array into a string
+};
```
Comment (sjproctor) [inline code-challenges.test.js:49]: Beautiful!

File: code-challenges.test.js [hunk 3]
```diff
@@ -10,52 +12,117 @@
...
+//return the new array with the words included
 
+const wordsWith = (arr, str) => {
+  newArr = arr.filter((value) => {
+    if (value.includes(str)) {
+      return value;
+    }
+  });
+  return newArr;
+};
```
Comment (sjproctor) [inline code-challenges.test.js:90]: Filter is def the right move here! The cool thing about filter is it has a built in `if` statement. So you can shorten your code a bit.
```
arr.filter((value) => {
    return value.includes(str) {
    }
```

File: code-challenges.test.js [hunk 4]
```diff
@@ -10,52 +12,117 @@
...
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
+const hand1 = [5, 5, 5, 3, 3];
 // Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
+const hand2 = [5, 5, 3, 3, 4];
 // Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
+const hand3 = [5, 5, 5, 5, 4]; // wait, this is not a full house?
```
Comment (sjproctor) [inline code-challenges.test.js:100]: Yeah a full house is exactly three of a kind and two of a kind.

File: code-challenges.test.js [hunk 5]
```diff
@@ -10,52 +12,117 @@
...
+    expect(leet(hand2)).toEqual(false);
+    expect(leet(hand3)).toEqual(false);
+  });
+});
 
 // b) Create the function that makes the test pass.
+
+// i dont know how to play cards, i googled and i think this is the rules in poker
+// full house is 3 of the same kind and a pair
+// what if the hand is more than 5???
```
Comment (sjproctor) [inline code-challenges.test.js:115]: Great question, thinking about edge cases is always a good idea.

File: code-challenges.test.js [hunk 6]
```diff
@@ -10,52 +12,117 @@
...
+
+  for (var value of arr) {
+    //iterating over the array
+    if (count[value]) {
+      count[value] += 1;
+    } else {
+      count[value] = 1;
+    }
+  }
+};
```
Comment (sjproctor) [inline code-challenges.test.js:128]: If you are unable to get your code working a good next step is to have a thorough explanation of your thought process.

File: interview-questions.md [hunk 7]
```diff
@@ -7,53 +8,44 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a block in Ruby?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: These are chunks of code enclosed within curly braces. The name of the block is assigned always finishes of with end
```
Comment (sjproctor) [inline interview-questions.md:11]: A block can be encased in curly braces or a do/end in Ruby.

File: interview-questions.md [hunk 8]
```diff
@@ -7,53 +8,44 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
+Researched answer:Other kinds of databases are NoSQL, Cloud, Columnar, Wide-Column, Object-Oriented, Key-Value, Heirarchical, Document, Graph, and Time Series Database. Relational Database is a digital databased on the relational model of data. It is maintained by RDBMS (Relatonal Database Management System)
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: Primary keys are assigned to identify each data collection. They are important because you can call on the entire data in a row by referencing it's primary key.
```
Comment (sjproctor) [inline interview-questions.md:35]: Great! They also ensure each entry is unique.

File: ruby-challenges.rb [hunk 9]
```diff
@@ -10,6 +10,15 @@
...
 
+def get_words_with(array, str)
+  getWords = []
+  array.each { |value| getWords.push(value) if value.include?(str) }
+  p getWords
+end
+get_words_with(beverages_array, letter_o)
+get_words_with(beverages_array, letter_t)
+
+# NOTE: the syntax on line 15 got automatically changed by prettier extension. It's kinda cool that you can write iterables and if statements like this
```
Comment (sjproctor) [inline ruby-challenges.rb:21]: That is super fun!

File: ruby-challenges.rb [hunk 10]
```diff
@@ -19,17 +28,56 @@
 nums_array2 = [25, 17, 47, 11]
 # Expected output: 100
 
+def sum(array)
+  array.sum
+end
 
+p sum nums_array1
+p sum nums_array2
```
Comment (sjproctor) [inline ruby-challenges.rb:36]: Beautiful! I love how concise this is.

File: ruby-challenges.rb [hunk 11]
```diff
@@ -19,17 +28,56 @@
...
+
+  def faster(num)
+    @current_speed += num
+  end
+
+  def slower(num)
+    @current_speed -= num
+    if @current_speed <= 0
+      @current_speed = 0
+      p 'You are stopped.'
```
Comment (sjproctor) [inline ruby-challenges.rb:69]: Due to Ruby's implicit return you can just have the `p` on the method call and get the string on line 69 as a return if the condition described on line 67 is met.

**Review by sjproctor — APPROVED** [review]
Wonderful job Mikka. Make sure you are putting the test variables inside the scope of the test for your JavaScript section!
